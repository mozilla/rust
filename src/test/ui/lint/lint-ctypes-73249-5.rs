// revisions: min_tait full_tait
#![feature(min_type_alias_impl_trait)]
#![cfg_attr(full_tait, feature(type_alias_impl_trait))]
//[full_tait]~^ WARN incomplete
#![deny(improper_ctypes)]

pub trait Baz { }

impl Baz for u32 { }

type Qux = impl Baz;

fn assign() -> Qux { 3 }

#[repr(transparent)]
pub struct A {
    //~^ ERROR type alias impl traits are not allowed as field types in structs
    x: Qux,
}

extern "C" {
    pub fn lint_me() -> A;
}

fn main() {}
