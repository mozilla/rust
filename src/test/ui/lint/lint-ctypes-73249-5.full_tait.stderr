warning: the feature `type_alias_impl_trait` is incomplete and may not be safe to use and/or cause compiler crashes
  --> $DIR/lint-ctypes-73249-5.rs:3:32
   |
LL | #![cfg_attr(full_tait, feature(type_alias_impl_trait))]
   |                                ^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(incomplete_features)]` on by default
   = note: see issue #63063 <https://github.com/rust-lang/rust/issues/63063> for more information

error: type alias impl traits are not allowed as field types in structs
  --> $DIR/lint-ctypes-73249-5.rs:16:1
   |
LL |   type Qux = impl Baz;
   |   -------------------- type alias defined here
...
LL | / pub struct A {
LL | |
LL | |     x: Qux,
   | |     ------ this field contains a type alias impl trait
LL | | }
   | |_^

error: aborting due to previous error; 1 warning emitted

